File: API/Entities/Member.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Member
{
    public string Id { get; set; } = null!;
    public DateTime Created { get; set; } = DateTime.UtcNow;
    public DateOnly DateOfBirth { get; set; }
    public string? Description { get; set; }
    public required string DisplayName { get; set; }
    public required string Gender { get; set; }
    public DateTime LastActive { get; set; } = DateTime.UtcNow;
    public string? ImageUrl { get; set; }
    public required string City { get; set; }
    public required string Country { get; set; }

    [ForeignKey(nameof(Id))]
    [JsonIgnore]
    public AppUser User { get; set; } = null!;

    [JsonIgnore]
    public IList<Photo> Photos { get; set; } = [];

}
    

File: client/src/app/models/member.ts
import { Gender } from "./gender";

export type Member = {
    id:string
    displayName:string
    dateOfBirth:string
    imageUrl?:string
    created:string
    lastActive:string
    gender:string
    description?:string
    country:string
    city:string
}

export class MemberSearchParams {
    
    constructor (pageIndex:number, pageSize:number, gender?:Gender) {
        this.pageIndex = pageIndex;
        this.pageSize = pageSize;
        this.gender = gender;
        this.orderBy = "displayName";
    }
    
    gender?: Gender
    minAge = 18
    maxAge = 100
    pageIndex = 0;
    pageSize = 10;
    orderBy = "";
}

export class MemberLikesSearchParams {
    pageIndex = 0;
    pageSize = 10;
    predicate:"liked"|"likedBy"|"mutual" = "likedBy";
}